src/coreclr/jit/assertionprop.cpp

```diff
@@ -164,6 +164,35 @@ bool IntegralRange::Contains(int64_t value) const
         case GT_CAST:
             return ForCastOutput(node->AsCast());
 
+#if defined(FEATURE_HW_INTRINSICS)
+        case GT_HWINTRINSIC:
+            switch (node->AsHWIntrinsic()->GetHWIntrinsicId())
+            {
+#if defined(TARGET_XARCH)
+                case NI_BMI1_TrailingZeroCount:
+                case NI_BMI1_X64_TrailingZeroCount:
+                case NI_LZCNT_LeadingZeroCount:
+                case NI_LZCNT_X64_LeadingZeroCount:
+                case NI_POPCNT_PopCount:
+                case NI_POPCNT_X64_PopCount:
+#elif defined(TARGET_ARM64)
+                case NI_AdvSimd_PopCount:
+                case NI_AdvSimd_LeadingZeroCount:
+                case NI_AdvSimd_LeadingSignCount:
+                case NI_ArmBase_LeadingZeroCount:
+                case NI_ArmBase_Arm64_LeadingZeroCount:
+                case NI_ArmBase_Arm64_LeadingSignCount:
+#else
+#error Unsupported platform
```

Comment:
Does it make much sense to complicate life for new platforms here? I'd just use `break` here

Reply:
I saw that #error was used thoughout `hwintrinsic.h` for such ifdefs, but I get your point

Reply:
@EgorBo, IMO yes.

This is already behind the FEATURE_HW_INTRINSICS flag and so it's not something a nre platform had to consider in the first place.


Once a new platform is adding SIMD support, however, it's nicer (IMO) for all the places they should be touching up and handling to get flagged so they can iteratively work through them and ensure the quality is consistent with what everything else supports.

Worst case, they change this to an explicit `// TODO-TheirArch` so they can explicitly track it to be handled later. Without the check it's something that may easily get missed or forgotten about since